conceptual/Npgsql/types/json.md

```diff
@@ -0,0 +1,208 @@
+
+1. As simple strings
+2. As strongly-typed user-defined types (POCOs)
+3. As System.Text.Json DOM types (JsonDocument or JsonElement, [see docs](https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/use-dom#use-jsondocument))
+4. High-performance JSON parsing with [Utf8JsonReader](https://learn.microsoft.com/dotnet/standard/serialization/system-text-json/use-utf8jsonreader)
+5. Newtonsoft Json.NET
+
+## String mapping
+
+The simplest form of mapping to JSON is as a regular .NET string:
+
+```c#
+// Write a string to a json column:
+await using var command1 = new NpgsqlCommand("INSERT INTO test (data) VALUES ($1)", conn)
+{
+    Parameters = { new() { Value = """{ "a": 8, "b": 9 }""", NpgsqlDbType = NpgsqlDbType.Jsonb } }
+};
+await command1.ExecuteNonQueryAsync();
+
+// Read jsonb data as a string:
+await using var command2 = new NpgsqlCommand("SELECT data FROM test", conn);
+await using var reader = await command2.ExecuteReaderAsync();
+while (await reader.ReadAsync())
+{
+    Console.WriteLine(reader.GetString(0));
+}
+```
+
+> [!NOTE]
+> Note that when writing a string parameter as `jsonb`, you must specify `NpgsqlDbType.Jsonb`, otherwise Npgsql sends a `text` parameter which is incompatible with JSON.
+
+With this mapping style, you're fully responsible for serializing/deserializing the JSON data yourself (e.g. with System.Text.Json) - Npgsql simply passes your strings to and from PostgreSQL.
+
+## POCO mapping
+
+> [!WARNING]
+> As of Npgsql 8.0, POCO mapping is incompatible with NativeAOT. We plan to improve this, [please upvote this issue if you're interested](https://github.com/npgsql/npgsql/issues/5355).
+
+If your column JSON contains documents with a stable schema, you can map them to your own .NET types (or POCOs). The provider will use System.Text.Json APIs under the hood to serialize instances of your types to JSON documents before sending them to the database, and to deserialize documents coming back from the database. This effectively allows mapping an arbitrary .NET type - or object graph - to a single column in the database.
+
+Startin with Npgsql 8.0, to use this feature, you must first enable it by calling <xref:Npgsql.INpgsqlTypeMapperExtensions.EnableDynamicJson> on your <xref:Npgsql.NpgsqlDataSourceBuilder>, or, if you're not yet using data sources, on `NpgsqlConnection.GlobalTypeMapper`:
+
+### [NpgsqlDataSource](#tab/datasource)
+
+> [!NOTE]
+> `NpgsqlDataSource` was introduced in Npgsql 7.0, and is the recommended way to manage type mapping. If you're using an older version, see the other methods.
+
+```c#
+var dataSourceBuilder = new NpgsqlDataSourceBuilder(...);
+dataSourceBuilder.EnableDynamicJson();
+await using var dataSource = dataSourceBuilder.Build();
+```
+
+### [Global mapping](#tab/global)
+
+If you're using an older version of Npgsql which doesn't yet support `NpgsqlDataSource`, you can configure mappings globally for all connections in your application:
```

Comment:
Isn't that only required starting with 8.0, as 7.0 and 6.0 can just send them as is?

Reply:
There's a note above: "Startin with Npgsql 8.0, to use this feature, you must fir eanble it by calling..." (typo with Startin...). Do you think that's not enough?

Reply:
I don't understand what's the point of this sentence:

```

Reply:
Ah I see, OK - will remove it.

Reply:
Or rather tweak it - there are people using Npgsql 8.0 but who are still not using NpgsqlDataSource

Reply:
Done in #316